File: Sample Apps /Java/SampleIndoorLocationReporting/app/src/main/java/com/mist/samplelocationbackground/service/LocationJobService.java
package com.mist.samplelocationbackground.service;

import static com.mist.samplelocationbackground.Constants.ORG_SECRET;

import android.app.Application;
import android.app.job.JobParameters;
import android.app.job.JobService;
import android.content.Context;
import android.content.SharedPreferences;
import android.util.Log;

import com.mist.samplelocationbackground.MainActivity;
import com.mist.samplelocationbackground.handler.SDKCallbackHandler;
import com.mist.samplelocationbackground.initializer.MistSdkManager;

/**
 * LocationJobService This class implements logic for location job service which is used by job
 * scheduler. This class Handles system broadcast event to restart the location service
 */
public class LocationJobService extends JobService {
    private static boolean needReschedule = true;

    @Override
    public boolean onStartJob(JobParameters params) {
        startWorkOnNewThread();
        return true;
    }

    private void startWorkOnNewThread() {
        new Thread(new Runnable() {
            @Override
            public void run() {
                doWork();
            }
        }, "LocationJobService").start();
    }

    private void doWork() {
        MistSdkManager mistSdkManager = MistSdkManager.getInstance((Application) getApplication());
        SDKCallbackHandler sdkCallbackHandler = new SDKCallbackHandler(getApplicationContext());
        mistSdkManager.init(ORG_SECRET, sdkCallbackHandler);
        mistSdkManager.startMistSDK();
        Log.d("TAG", "SampleLocationApp: doWork() ThreadName: " + Thread.currentThread().getName());
    }

    @Override
    public boolean onStopJob(JobParameters params) {
        jobFinished(params, needReschedule);
        return needReschedule;
    }

    public static void needJobReschedule(boolean needReschedule) {
        LocationJobService.needReschedule = needReschedule;
    }
}

File: Sample Apps /Kotlin/SampleIndoorLocationReporting/app/src/main/java/com/example/sampleindoorlocationreporting/service/LocationJobService.kt
package com.example.sampleindoorlocationreporting.service

import android.app.job.JobParameters
import android.app.job.JobService
import android.util.Log
import com.example.sampleindoorlocationreporting.Constants
import com.example.sampleindoorlocationreporting.handler.SDKCallbackHandler
import com.example.sampleindoorlocationreporting.initializer.MistSdkManager

class LocationJobService : JobService() {

    private val constants = Constants()
    private var needReSchedule : Boolean = true
    override fun onStartJob(p0: JobParameters?): Boolean {
        startWorkOnNewThread()
        return true
    }

    private fun startWorkOnNewThread() {
        Thread({ doWork() }, "LocationJobService").start()
    }

    private fun doWork() {
        val mistSdkManager = MistSdkManager()
        mistSdkManager.getInstance(application)
        val sdkCallbackHandler = SDKCallbackHandler(applicationContext)
        mistSdkManager.init(constants.orgSecret, sdkCallbackHandler, constants.orgId)
        mistSdkManager.startMistSDK()
        Log.d("TAG","SampleLocationApp: doWork() ThreadName: " + Thread.currentThread().name)

    }

    override fun onStopJob(p0: JobParameters?): Boolean {
        jobFinished(p0,needReSchedule)
        return needReSchedule
    }

    fun needJobReschedule(reschedule: Boolean) {
        LocationJobService().needReSchedule = reschedule
    }
}